FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m

```diff
@@ -1164,6 +1207,95 @@ - (void)testConfigureConfigWithValidInput {
   XCTAssertEqual([config valueForKey:@"_appName"], kFIRDefaultAppName);
 }
 
+#pragma mark app extension.
+- (void)testETagAndFetchTimeUpdatedInSharedUserDefaultsForAppExtension {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+  for (int i = 0; i < 1; i++) {
+    expectations[i] = [self
+        expectationWithDescription:
+            [NSString stringWithFormat:@"Test fetch configs for app extension - instance %d", i]];
+    XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusNoFetchYet);
+    FIRRemoteConfigFetchCompletion fetchCompletion = ^void(FIRRemoteConfigFetchStatus status,
+                                                           NSError *error) {
+      XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusSuccess);
```

Comment:
Nit: this is a pretty big completion handler and a lot going on here, some comments splitting things up into sections with an explanation of why it's expected would be very helpful for future readers. Note: I've only added comments and re-arranged two calls. Example:

```
// Fetch should be successful, with no errors.
XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusSuccess);
XCTAssertNil(error);

// A fetch occurred, so ensure `lastFetchTime` is valid and non-zero.
XCTAssertNotNil(_configInstances[i].lastFetchTime);
XCTAssertGreaterThan(_configInstances[i].lastFetchTime.timeIntervalSince1970, 0,
                           @"Last fetch time interval should be set.");

// The app and extension share data through UserDefaults using the `appGroupID` as the
// suite name.
NSUserDefaults *externalUserDefaults =
          [[NSUserDefaults alloc] initWithSuiteName:[self firstAppOptions].appGroupID];

// Validate the ETag is correct. 
NSString *currentETag = [externalUserDefaults objectForKey:@"frc.latestETag"];
XCTAssertTrue([currentETag isEqualToString:@"etag1-0"]);

// Validate the last fetch time is correct.
NSNumber *lastFetchTime = [externalUserDefaults objectForKey:@"frc.lastSuccessfulFetchTime"];
NSDate *lastFetchedDateTime = _configInstances[i].lastFetchTime;
XCTAssertEqual(lastFetchTime.doubleValue, [lastFetchedDateTime timeIntervalSince1970]);

// Update the eTag
XCTAssertTrue([currentETag isEqualToString:@"etag1-0"]);
[expectations[i] fulfill];
```

Reply:
(Also relevant in the second large block added below)

Reply:
Done.